tasks/Task.cpp

```diff
@@ -78,12 +74,10 @@ void Task::updateHook()
     }
 
     //grab the frame
-    Frame *frame = vizkit3dWorld->grabFrame();
-    buffer.init(*frame);
-
-    //write in the output frame port
-    cameraFrame.reset(&buffer);
-    _frame.write(cameraFrame);
+    std::auto_ptr<Frame> frame(new Frame());
+    vizkit3dWorld->grabFrame(*frame.get());
+    frame->time = cameraPose.time;
```

Comment:
Actually, this would work since read(data) will fill data with the last sample received if there is no new data. My mistake.